RetailCoder.VBE/Common/TimerHook.cs

```diff
@@ -31,8 +33,27 @@ public void Attach()
             try
             {
                 var timerId = (IntPtr)Kernel32.GlobalAddAtom(Guid.NewGuid().ToString());
```

Comment:
So I'm reading the documentation on the [SetTimer](https://msdn.microsoft.com/en-us/library/windows/desktop/ms644906(v=vs.85).aspx) and there is no mention of needing a global atom. I'm not sure what purpose the global atom is supposed to serve here, given that we are generating a new GUID every time we instantiate the class which will never match any existing entries in the global atom table. Can anyone explain why we need this? Otherwise, this strikes me as a cargo-cult programming...

Reply:
+1, although you need to supply a timer ID of some sort.  The relevant part of the documentation says "If the hWnd parameter is not NULL and the window specified by hWnd already has a timer with the value nIDEvent, then the existing timer is replaced by the new timer".  The rest of it (e.g. saying that a new timer ID will be allocated for you) refers to where hWnd is being passed a NULL, so doesn't seem to apply.  

Using a global atom is clearly not a requirement though.  You just need to be prepared for your specified ID to be already assigned to a timer (e.g. by another addin, or the VBE itself), and try another if it fails (which you should do even with the global atom anyway).